File: Python3/Best Time to Buy and Sell Stock II.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        long = True
        
        for i in range(1, len(prices)):
            if prices[i-1] < prices[i]:
                if long:
                    profit -= prices[i-1]
                    long = not long
                else:
                    continue
            else:
                if long:
                    continue
                else:
                    profit += prices[i-1]
                    long = not long
       
        if not long:
            profit += prices[len(prices) - 1]
            
        return profit

File: JavaScript/Best Time to Buy and Sell Stock II.js
var maxProfit = function(prices) {
        let profit = 0
        let long = true
        
        for (let i =1; i < prices.length; i++) {
            if (prices[i-1] < prices[i]) {
                if (long) {
                    profit -= prices[i-1]
                    long = !long
                }
            } else {
                if (!long) {
                    profit += prices[i-1]
                    long = !long
                }
            }
        }
        if (!long) {
            profit += prices[prices.length - 1]
        }    
        return profit
}
